### DISCUSSION TITLE
Differences in Behavior Between Custom Go Build and Official PocketBase Executables

### DISCUSSION THREAD

**Discussion by Rfluegel1** [pocketbase/pocketbase#4187] [Q&A] [answered]
Hello everyone,

I'm relatively new to Go and have recently started working with PocketBase ([link to PocketBase](https://pocketbase.io/docs/go-overview/)). I've taken an initiative to extend PocketBase's functionality for my project. To do this, I've copied the main.go file from the PocketBase Go SDK and built my own executable. The build process seems to go smoothly, and I've managed to replicate most of the behaviors successfully.

However, I'm observing some differences in the behavior of my custom-built executable compared to the official macOS executable available on the [PocketBase website](https://pocketbase.io/docs/). My understanding is that all these executables are built from the same main.go file but are compiled with different architecture options.

I'm curious if anyone here has experience with building PocketBase from source and could shed some light on whether there are any differences in the source main.go file used for the official builds. Also, if anyone could point me to the source code for the main.go file that is used to create these official executables, it would be incredibly helpful.

Specifically, I'm looking to understand:

1. If the official builds use a different main.go or additional settings.
2. Any common pitfalls or considerations when building PocketBase from source.
3. How to ensure that my custom build replicates the behavior of the official builds as closely as possible.

Any insights, tips, or references to relevant documentation would be greatly appreciated!

Thank you in advance for your help!

**Comment by ganigeorgiev** [comment, marked as answer]
All prebuilt executables are based on the [`example/base/main.go` file](https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go).

> Any common pitfalls or considerations when building PocketBase from source.

If you are cross-compiling make sure that `CGO_ENABLED=0` (eg. `CGO_ENABLED=0 go build`).

> How to ensure that my custom build replicates the behavior of the official builds as closely as possible.

Check the above `example/base/main.go` file but note that you may not need everything that is available there. For example, if you are already extending PocketBase with Go, you may not need the JSVM plugin registration if you are not planning to extend your compiled program with JS.

**Reply by Rfluegel1** [reply]
Copying over the example/base/main.go wholly fixed the immediate behavior differences that I've noticed. I will have to take a closer look to see what is just cruft for my use case. Thank you much!
